tox.ini

```diff
@@ -24,18 +24,22 @@ test_dir =
     act: cli/action
     visprompt: cli/visual_prompting
 deps =
-    py38:  torch @ https://download.pytorch.org/whl/cu117/torch-1.13.1%2Bcu117-cp38-cp38-linux_x86_64.whl
-    py38:  torchvision @ https://download.pytorch.org/whl/cu117/torchvision-0.14.1%2Bcu117-cp38-cp38-linux_x86_64.whl
-    py38:  torchtext @ https://download.pytorch.org/whl/torchtext-0.14.1-cp38-cp38-linux_x86_64.whl
-    {all,act,cls,det,seg,iseg,iseg_t}-py38: mmcv-full @ https://download.openmmlab.com/mmcv/dist/cu117/torch1.13.0/mmcv_full-1.7.0-cp38-cp38-manylinux1_x86_64.whl
-    py39:  torch @ https://download.pytorch.org/whl/cu117/torch-1.13.1%2Bcu117-cp39-cp39-linux_x86_64.whl
-    py39:  torchvision @ https://download.pytorch.org/whl/cu117/torchvision-0.14.1%2Bcu117-cp39-cp39-linux_x86_64.whl
-    py39:  torchtext @ https://download.pytorch.org/whl/torchtext-0.14.1-cp39-cp39-linux_x86_64.whl
-    {all,act,cls,det,seg,iseg,iseg_t}-py39:  mmcv-full @ https://download.openmmlab.com/mmcv/dist/cu117/torch1.13.0/mmcv_full-1.7.0-cp39-cp39-manylinux1_x86_64.whl
-    py310: torch @ https://download.pytorch.org/whl/cu117/torch-1.13.1%2Bcu117-cp310-cp310-linux_x86_64.whl
-    py310: torchvision @ https://download.pytorch.org/whl/cu117/torchvision-0.14.1%2Bcu117-cp310-cp310-linux_x86_64.whl
-    py310: torchtext @ https://download.pytorch.org/whl/torchtext-0.14.1-cp310-cp310-linux_x86_64.whl
-    {all,act,cls,det,seg,iseg,iseg_t}-py310: mmcv-full @ https://download.openmmlab.com/mmcv/dist/cu117/torch1.13.0/mmcv_full-1.7.0-cp310-cp310-manylinux1_x86_64.whl
+    py38-pt1:  torch @ https://download.pytorch.org/whl/cu117/torch-1.13.1%2Bcu117-cp38-cp38-linux_x86_64.whl
+    py38-pt1:  torchvision @ https://download.pytorch.org/whl/cu117/torchvision-0.14.1%2Bcu117-cp38-cp38-linux_x86_64.whl
+    {all,act,cls,det,seg,iseg,iseg_t}-py38-pt1: mmcv-full @ https://download.openmmlab.com/mmcv/dist/cu117/torch1.13.0/mmcv_full-1.7.0-cp38-cp38-manylinux1_x86_64.whl
```

Comment:
It's just a question. mmcv for pt2 isn't added in `deps`, is it installed well automatically?

Reply:
They do not provide the pre-built wheels for mmcv1.x & torch2.x combination, so that we need to build it from PyPI source package.
`pip install otx` or something similar behind the tox build does it.